scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -5099,14 +5101,13 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     stats.toList
   }
 
-  def topStatSeq(): List[Stat] = statSeq(topStat, errorMsg = "expected class or object definition")
```

Comment:
why not just redefine it as
```
statSeq(
   if (dialect.allowToplevelTerms) consumeStat else topStat,
   "some appropriate error message"
)
```
and remove most of the method parameter changes?

Reply:
I didn't like that because then the name is misleading, since it might permit things that are not top-level statements (as defined by `topStat` and the Scala 3 spec). I deleted it precisely so I would have to audit every position to see whether it really wanted a topStat or the dialect-specific notion of one. 
I would suggest leaving it as is I have it, or renaming `topStat` to be `scala3TopStat`. What do you prefer?

Reply:
i would recommend considering this rename idea as part of some other pull request
 this one is about package, so let's stick to that.

Reply:
Okay sure. I'll just leave a comment warning about the confusion of topStatSeq() not always meaning a Seq of topStat.

Reply:
oh well...

Reply:
@adampauls would you consider revising this like i suggested? it may have been a mistake to say "Okay sure" and yet leave this unchanged.

here's an example of the adverse effect: https://github.com/scalacenter/scalafix/pull/1552#pullrequestreview-882591159